### PR TITLE
#600: Add proxy support for sFTP

### PR DESCRIPTION
**Pull request by tonymurphy**
@juanjovazquez It would be good if sftp supported proxy server

### CODE REVIEW

File: ftp/src/main/scala/akka/stream/alpakka/ftp/impl/SftpOperations.scala [hunk 1]
```diff
@@ -31,7 +32,13 @@ private[ftp] trait SftpOperations { _: FtpLike[SSHClient, SftpSettings] =>
...
       knownHosts.foreach(path => ssh.loadKnownHosts(new File(path)))
 
-    ssh.connect(host.getHostAddress, port)
+    connectionSettings.proxy match {
+      case Some(p) =>
+        ssh.setSocketFactory(new DefaultSocketFactory(p))
+        ssh.connect(host.getHostAddress, port)
+      case None =>
+        ssh.connect(host.getHostAddress, port)
+    }
```
Comment (juanjovazquez) [inline ftp/src/main/scala/akka/stream/alpakka/ftp/impl/SftpOperations.scala]: Some code duplication here. Probably just an `if` might work.

**Comment by juanjovazquez** [comment]
Apparently the new test is not working on travis.

**Comment by ennru** [comment]
Scalaformat is complaining:
Unformatted code found. Please run 'test:compile' and commit the reformatted code.

**Comment by ennru** [comment]
Fixes #600

**Comment by ennru** [comment]
We'll bake a release in two weeks time and it would be great, if we could get your contribution in.

**Comment by tonymurphy** [comment]
Will try to get those tests working asap. Struggling with the proxy server. Tried to use akka http, but unfortunately it doesn't yet support CONNECT. Thanks!

**Comment by ennru** [comment]
@tonymurphy any chance you get back to this?

**Comment by tonymurphy** [comment]
@ennru Unfortunately I ran into problems. There is some handshaking / state issues setting up connections between Apache sshd and littleproxy. One of the 3 tests will work. Due to other work commitments, I had to park it for the moment. I do hope to revisit in the coming weeks

`21:31:11.996 DEBUG rg.littleshoot.proxy.impl.ProxyToServerConnection] - (DISCONNECTED {tunneling}) [id: 0x6937f1d9, L:/127.0.0.1:49945 ! R:/127.0.0.1:21000]: Disconnected
21:31:11.996 DEBUG org.apache.sshd.common.io.nio2.Nio2ServiceFactory] - Group successfully shut down
21:31:11.997 DEBUG org.apache.sshd.common.io.nio2.Nio2ServiceFactory] - close(org.apache.sshd.common.io.nio2.Nio2ServiceFactory@cedee22)[Immediately] closed
21:31:11.997 ERROR [net.schmizz.sshj.transport.TransportImpl]         - Received end of connection, but no identification received. 
 `

**Comment by ennru** [comment]
This PR hasn't been touched for a very long time.
Please re-open or create a new to continue this work.

**Pull request closed without merging**